### ISSUE TITLE
External CSS dependencies for PROD

### ISSUE THREAD

**Issue by ggaborx** [mgechev/angular2-seed#627]
I'm using n2-material package in my project. It works fine in DEV build but .css files missing from the PROD build. I don't know if it was a bug or I do it wrong. So here is how I added them in my project.config.ts:
```
constructor() {
    ...
    let additional_deps = [
         { src: 'ng2-material/dist/ng2-material.css', inject: true },
         { src: 'ng2-material/dist/font.css', inject: true }
    ];

    this.DEV_NPM_DEPENDENCIES = this.DEV_NPM_DEPENDENCIES.concat(normalizeDependencies(additional_deps));
    this.PROD_NPM_DEPENDENCIES = this.PROD_NPM_DEPENDENCIES.concat(normalizeDependencies(additional_deps));

    this.APP_ASSETS = [];

    this.DEV_DEPENDENCIES = this.DEV_NPM_DEPENDENCIES.concat(this.APP_ASSETS);
    this.PROD_DEPENDENCIES = this.PROD_NPM_DEPENDENCIES.concat(this.APP_ASSETS);
    ...
}
```

I found that the old and removed build.html_css.prod.ts task's getExternalCss() function used the PROD_DEPENDENCIES instead of the APP_ASSETS:
```
function getExternalCss() {
  return PROD_DEPENDENCIES.filter(d => /\.css$/.test(d.src));
}
```
But in the new merged one (build.html_css.ts) it uses the APP_ASSETS:
```
function getExternalCss() {
  return APP_ASSETS.filter(d => /\.css$/.test(d.src));
}
```

Because currently I have no any APP_ASSETS I just replaced that func with the old one to check results and required n2-material .css files bundled well.

**Comment by d3viant0ne** [comment]
iirc this was all removed to all the tools to be updated without causing issues with what people build on top of the seed.

Your prod build should go in the project section so it can be tailored to the application you are building.

//cc @ludohenin

**Comment by ggaborx** [comment]
I don't really understand. What was removed? I already updated my copy of the seed so I also won't use the removed build.http_css.prod.ts anymore. I'm using all original tasks and gulpfile. Or do you mean I should create my own build.http_css.ts task file to merge third party css as part of the prod deps? All config above part of my project.config.ts. I saw in some forks that people put all third party css files into the assets folder then loaded them from there. Is this the preferred way?

**Comment by ggaborx** [comment]
I spent some time to find which config values used in which seed tasks and I found the following:

- APP_ASSETS used in build.html_css.ts and cs-lint.ts and filtered only for .css files. So it should be renamed to APP_CSS or something similar. 

- DEV_DEPENDENCIES used in build.index.dev.ts and no any type filter applied so it would process .js and .css files. It contains the whole APP_ASSETS but the build.dev task processes that independently (via build.html_css) then executes it. 

- PROD_DEPENDENCIES used in build.bundles.ts and filtered only for .js files. It also contains the whole APP_ASSETS. 

So if I understood corretly:
- APP_ASSETS is only for project's local .css files.
- DEV/PROD_NPM_DEPENDECIES are for only for NPM .js files
- DEV/PROD_DEPENDECIES contains both above but filtered in tasks for .css/.js types.
- NPM deps always added by using normalizeDependencies(...) function which will resolve node_modules/* paths.
- APP_ASSETS don't use the normalizeDependencies(...) so it shouldn't contain .css from NPM.

How should I add third party .css/.font (for example bootstrap or font-awesome) dependencies into the project.config?
